cmd/create/cluster/cmd.go

```diff
@@ -964,15 +966,28 @@ func run(cmd *cobra.Command, _ []string) {
 					os.Exit(1)
 				}
 				selectedARN := ""
-				expectedResourceIDForAccRole := strings.ToLower(fmt.Sprintf("%s-%s-Role", rolePrefix, role.Name))
 				for _, rARN := range roleARNs {
 					resourceId, err := aws.GetResourceIdFromARN(rARN)
 					if err != nil {
 						r.Reporter.Errorf("Failed to get resource ID from arn. %s", err)
 						os.Exit(1)
 					}
 					lowerCaseResourceIdToCheck := strings.ToLower(resourceId)
-					if lowerCaseResourceIdToCheck == expectedResourceIDForAccRole {
+
+					// Searches for outdated worker role and prompts users to delete and recreate
+					if lowerCaseResourceIdToCheck == getOutdatedWorkerRole(isHostedCP, rolePrefix) {
```

Comment:
I'm not sure if we need this, creating roles with the `-HCP` infix is currently only available for QE and DEV.
Users creating roles during the public preview are using classic account roles without the `-HCP` prefix.

Reply:
hi @oriAdler 
For clarification:
Is it expected that users, after the public preview, will create hcp account roles with the `-HCP` infix in the account role names? (same as QE and DEV)

Reply:
Yes, for GA we are expecting it will be available for all users.
The user won't set the `-HCP` explicitly, but by choosing `create account-roles --hosted-cp` they will get a set with `-HCP` infix.